### ISSUE TITLE
[Request] Support for Building gadgets

### ISSUE THREAD

**Issue by Hund** [Vazkii/Botania#2685]
I asked for Building gadgets to support the Black hole talisman, but according to Direwolf20, the Black hole talisman doesn't use Forges built in inventory container and uses its own thing and it doesn't look like he wants to add support for it.

Would it be possible if Botania could support the Building gadgets instead? If that's possible? :)

**Comment by quat1024** [comment]
I looked in to this, here are my notes.

* The aforementioned "own thing" is IBlockProvider. The cobble/dirt/terraform use it, as well as the Black Hole Talisman and Hand of Ender.
* *I want IBlockProvider gone*. It hasn't been touched since its creation in 2015, and it feels very, *very* similar to what capabilities can do.
* Cutting down the amount of interfaces in the Botania API is also a good thing, considering how 1.13 cuts `@Optional` in favor of using capabilities, or something like that idk.
* However, IBlockProvider crucially provides access to the player entity that is requesting stuff.
* This is important so that mana can be subtracted from the player's inventory, discounts can be applied for manaweave armor, etc.
* But access to a player is not always available w/ item handler capabilities (nor should it be)

Tl;dr this problem is much more complicated than it looks considering it's not free to get itemstacks out of Rod of the Lands/Seas/whatever the cobble one is called/Hand of Ender.

The Black Hole Talisman, though, is more doable
